arch/arm/core/aarch32/isr_wrapper.S

```diff
@@ -183,6 +183,14 @@ _idle_state_cleared:
 	 * interface function.
 	 */
 	cpsie i
+
+	/*
+	 * Skip calling the isr if it is a spurious interrupt.
```

Comment:
Perhaps this comment is misleading, we don't really check for spurious interrupt, we just make sure we are not going out of bound which has the side effect of covering spurious interrupts from GIC.